n2y/plugins/render.py

```diff
@@ -0,0 +1,210 @@
+    if loaders is None:
+        loaders = [
+            jinja2.FileSystemLoader('.'),
+        ]
+
+    return jinja2.ChoiceLoader(loaders)
+
+
+def _generate_source_line_list(template, context):
+    generator = template.generate(**context)
+    source = ''.join(generator)
+    # template generator usually loses trailing new line.
+    if source and source[-1] != '\n':
+        source += '\n'
+    return source.splitlines(keepends=True)
+
+
+def _generate_output_lines(environment, source_line_list):
+    output_generator = (line for line in source_line_list)
+    post_process_filters = post_processing_filter_list(environment)
+    for post_process_filter in post_process_filters:
+        output_generator = (x for x in post_process_filter(output_generator))
+    return output_generator
+
+
+class RawFencedCodeBlock(FencedCodeBlock):
+    """
+    Adds support for raw codeblocks.
+
+    Any code block whose caption begins with "{=language}" will be made into a
+    raw block for pandoc to parse. This is useful if you need to drop into Raw
+    HTML or other formats.
+
+    See https://pandoc.org/MANUAL.html#generic-raw-attribute
+    """
+    trigger_regex = re.compile(r'^\{template\}')
+
+    def __init__(self, client, notion_data, page, get_children=True):
+        super().__init__(client, notion_data, page, get_children)
+        result = self.caption.matches(self.trigger_regex)
+        if not result or self.client.render_config is None:
+            raise UseNextClass()
+        if self.client.render_config is None:
+            raise NotImplementedError(
+                'The "render-config" argument must be set to use the "render" plugin'
+            )
+
+    def to_pandoc(self):
+        pandoc_language = self.notion_to_pandoc_highlight_languages.get(
+            self.language, self.language,
+        )
+        if pandoc_language not in self.pandoc_highlight_languages:
+            if pandoc_language != "plain text":
+                msg = 'Dropping syntax highlighting for unsupported language "%s" (%s)'
+                logger.warning(msg, pandoc_language, self.notion_url)
+            language = []
+        else:
+            language = [pandoc_language]
+        return self._render(CodeBlock(('', language, []), self.rich_text.to_plain_text()))
+
+    def _render(self, ast):
+        context = self.client.context_from_yaml_cache()
+        config = self.client.render_config
+        try:
+            id = str(uuid.uuid4())
+            output_name = f'{id}-output.md'
+            template_name = f'{id}-template.md'
+            open(output_name, 'x').close()
+            with open(template_name, 'x') as file:
+                file.write(pandoc.write(ast))
+            render_template_to_file(config, template_name, context, output_name)
+            with open(output_name, 'r') as file:
+                rendered_ast = pandoc.read(file.read())
+            return rendered_ast[1][0]
+        finally:
+            os.remove(output_name)
```

Comment:
Can we use the python `tempfile` module to create/delete these?

Reply:
yes